pkg/sql/sem/tree/normalize.go

```diff
@@ -978,10 +978,19 @@ func init() {
 	DecimalOne.SetInt64(1)
 }
 
-// ReType ensures that the given numeric expression evaluates
+// ReType ensures that the given expression evaluates
 // to the requested type, inserting a cast if necessary.
 func ReType(expr TypedExpr, wantedType *types.T) TypedExpr {
-	if wantedType.Family() == types.AnyFamily || expr.ResolvedType().Identical(wantedType) {
+	resolvedType := expr.ResolvedType()
+	if wantedType.Family() == types.AnyFamily || resolvedType.Identical(wantedType) {
+		return expr
+	}
+	// Casting to an "any" tuple is not allowed because there is no way to define
+	// the volatility in LookupCastVolatility.
+	if wantedType.Family() == types.TupleFamily &&
```

Comment:
can we not always use stable as a volatility? i think that's what Postgres does. in these circumstances.

Reply:
that sounds cool! will try.